app/assets/javascripts/preferences/panes/two-factor-auth/TwoFactorAuthView.tsx

```diff
@@ -0,0 +1,135 @@
+import { FunctionalComponent, FunctionComponent } from 'preact';
+import {
+  Title,
+  Text,
+  PreferencesGroup,
+  PreferencesSegment,
+} from '../../components';
+import { Switch } from '../../../components/Switch';
+import { observer } from 'mobx-react-lite';
+import { DecoratedInput } from '../../../components/DecoratedInput';
+import { IconButton } from '../../../components/IconButton';
+import { ScanQRCode } from './ScanQRCode';
+import { SaveSecretKey } from './SaveSecretKey';
+import { Verification } from './Verification';
+import { TwoFactorActivation, TwoFactorAuth } from './model';
+import { downloadSecretKey } from './download-secret-key';
+import { CircleProgress } from '@/components/CircleProgress';
+import { useState } from 'preact/hooks';
+import { useEffect } from 'react';
+
+export const TwoFactorAuthView: FunctionComponent<{
+  auth: TwoFactorAuth;
+}> = observer(({ auth }) => (
+  <PreferencesGroup>
+    <PreferencesSegment>
+      <div className="flex flex-row items-center">
+        <div className="flex-grow flex flex-col">
+          <Title>Two-factor authentication</Title>
+          <Text>
+            An extra layer of security when logging in to your account.
+          </Text>
+        </div>
+        <Switch
+          checked={auth.enabled !== false}
```

Comment:
Can auth.enabled be anything different from a boolean?

Reply:
Yes. And yes, it doesn't look good :) I'm in the process of changing that.

Reply:
It should be better now. Commit 1f75df8e6c5ab944f885face09ba067836916592.